### PR TITLE
Project setup and movie detail page with two components (movie shuffle and read more text) 

### PR DESCRIPTION
**Pull request by flipkoehler**

### CODE REVIEW

File: pages/filmEmpfehlung/index.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// This site is for showing a random movie, bassed on the passed movie Array
```
Comment (MerleMak) [inline pages/filmEmpfehlung/index.js:1]:
```suggestion
// This site is for showing a random movie, based on the passed movie Array
```

File: pages/filmEmpfehlung/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import { movieData } from "../../assets/data/movieData";
+import RandomMoviePicker from "../../components/RandomMoviePicker/RandomMoviePicker";
+
+export default function randomMovie() {
+  return (
+    <>
+      <RandomMoviePicker movieArray={movieData} />
```
Comment (MerleMak) [inline pages/filmEmpfehlung/index.js:9]: Its a minor thing but if you call the movieArray just "movies", the plural is already descriptive for an array. Only arrays are named in plural form :)

File: pages/filmEmpfehlung/index.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    </>
```
Comment (MerleMak) [inline pages/filmEmpfehlung/index.js:10]:
```suggestion
      <RandomMoviePicker movieArray={movieData} />
```
You only need fractions if you have two parent elements in the return statement :)

File: pages/_app.tsx [hunk 4]
```diff
@@ -1,6 +1,21 @@
-import '../styles/globals.css'
-import type { AppProps } from 'next/app'
+import "../styles/globals.css";
```
Comment (MerleMak) [inline pages/_app.tsx:1]:
```suggestion
```

File: pages/_app.tsx [hunk 5]
```diff
@@ -1,6 +1,21 @@
...
+  return (
+    <>
+      <GlobalStyles />
+      <Head>
+        <title>🎬 Movie Flip 🎬</title>
+        <meta
+          name="Movie Flip"
+          content="Die beste Adresse für schnelle Filmemfephlungen"
+        />
+        <link rel="icon" href="/favicon.ico" />
```
Comment (MerleMak) [inline pages/_app.tsx:16]: is it not better to use the Link component from next? Are you getting linting issues?

File: components/Header/Header.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+// this is my header component that will contain logo and menu later on
+import styled from "styled-components";
+
+export default function Header() {
+  return <H2Styled>Movie Flip</H2Styled>;
+}
+
+const H2Styled = styled.h2`
```
Comment (MerleMak) [inline components/Header/Header.js:8]: Naming convention would be StyledH2 😁

File: components/Main/Main.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+// this component will contain all the content components besides the Header and footer. For now it will only contain a short welcome text and a link for a better navigation.
+
+import Link from "next/link";
+
+export default function Header() {
+  return (
+    <>
+      <h1>Willkommen zu Movie Flip 🎬</h1>
+      <h2>Deine Filmempfehlung in vier einfachen Schritten</h2>
+      <Link href="/filmEmpfehlung">
```
Comment (MerleMak) [inline components/Main/Main.js:10]: If you prefer, you can write the information which is relevant for the user in German, but everything on the developer site, should be in english, to keep it coherent:)

```suggestion
      <Link href="/movieRecommendations">
```

File: components/Main/Main.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+        <button type="button" aria-label="Filmempfehlung ansehen">
+          Filmempfehlung ansehen!
+        </button>
```
Comment (MerleMak) [inline components/Main/Main.js:13]: A link shouldn't include button. See here https://caninclude.glitch.me/caninclude?child=button&parent=link
But you can always style a link like a button :)

File: components/RandomMoviePicker/RandomMoviePicker.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+import Header from "../../components/Header/Header";
+import styled from "styled-components";
+
+export default function RandomMoviePicker({ movieArray }) {
+  const [randomPick, setRandomPick] = useState(0); // picks a random number
+  const listItems = movieArray.map((movie) => <li key={movie.id}></li>); // map through the movie Array
+  const productionYear = movieArray[randomPick].release_date.slice(0, 4); // slices down the production year, e.g. 2014-12-04 > 2014
+  useEffect(() => {
+    setRandomPick(Math.floor(Math.random() * listItems.length));
+  }, [listItems.length]); // calculates a random number on a page load
```
Comment (MerleMak) [inline components/RandomMoviePicker/RandomMoviePicker.js:15]: amazing! 👍🏼

File: components/RandomMoviePicker/RandomMoviePicker.js [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+  // return the movie details bassed on the random number
+  return (
+    <>
+      <Header />
+      <SectionStyled aria-label="movie content">
+        <ImageWrap aria-label="Image Wrap Div">
```
Comment (MerleMak) [inline components/RandomMoviePicker/RandomMoviePicker.js:22]: you don't really need the aria-labels here, as it is not important for a user to know that this a div or movie content, they will know by the alt attribute in the image :)

File: components/RandomMoviePicker/RandomMoviePicker.js [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+            height={304}
+          />
+        </ImageWrap>
+        <h2>{movieArray[randomPick].title}</h2>
+        <MovieTagsUl>
+          <MovieTags>{movieArray[randomPick].runtime} Minuten</MovieTags>
+          <MovieTags>{movieArray[randomPick].genres[0].name}</MovieTags>
+          <MovieTags>Jahr: {productionYear}</MovieTags>
+        </MovieTagsUl>
+        <ReadMore aria-label="mehr lesen oder weniger lesen">
```
Comment (MerleMak) [inline components/RandomMoviePicker/RandomMoviePicker.js:36]:
```suggestion
        <ReadMore aria-label="read more or read less">
```
better to keep everything in english as the screen reader includes a translator :)
Later you could lift up your state and change the aria label depending on IsReadMore being true or false :)

File: components/ReadMoreText/ReadMoreText.js [hunk 12]
```diff
@@ -0,0 +1,30 @@
+// This component is for fold and unfold the text. If a user klicks on the textlink it shows more  or less text depending on the useState
+
+import { useState } from "react";
+import styled from "styled-components";
+
+// this sets set state true or false // the children is the passed text
+export const ReadMore = ({ children }) => {
+  const text = children;
+  const [isReadMore, setIsReadMore] = useState(true);
+  const toggleReadMore = () => {
```
Comment (MerleMak) [inline components/ReadMoreText/ReadMoreText.js:10]: why did you choose to declare a constant instead of a little function?

**Review by MerleMak — COMMENTED** [review]
🥵 36 files in the first review.. i will forgive you, as you had to clean up some stuff and most of the files are fonts hahaha but please try and keep your PRs and Userstories small in the future :D

Other than that you did an amazing job! I left some remarks concerning coherent language and accessibility.
But concerning your logic, it's really really cool!
Keep up the good work ☺️👍🏼

File: pages/filmEmpfehlung/index.js (hunk 1 above)
Comment (flipkoehler) [reply pages/filmEmpfehlung/index.js:1]: done.

File: pages/filmEmpfehlung/index.js (hunk 2 above)
Comment (flipkoehler) [reply pages/filmEmpfehlung/index.js:9]: done.

File: pages/filmEmpfehlung/index.js (hunk 3 above)
Comment (flipkoehler) [reply pages/filmEmpfehlung/index.js:10]: done.

File: pages/_app.tsx (hunk 4 above)
Comment (flipkoehler) [reply pages/_app.tsx:1]: done.

File: pages/_app.tsx (hunk 5 above)
Comment (flipkoehler) [reply pages/_app.tsx:16]: done. It seems to work :)

File: components/Main/Main.js (hunk 7 above)
Comment (flipkoehler) [reply components/Main/Main.js:10]: done.

File: components/Main/Main.js (hunk 8 above)
Comment (flipkoehler) [reply components/Main/Main.js:13]: good to know :) 
done.

File: components/RandomMoviePicker/RandomMoviePicker.js (hunk 10 above)
Comment (flipkoehler) [reply components/RandomMoviePicker/RandomMoviePicker.js:22]: done.

File: components/RandomMoviePicker/RandomMoviePicker.js (hunk 11 above)
Comment (flipkoehler) [reply components/RandomMoviePicker/RandomMoviePicker.js:36]: done.

**Pull request merged**